frontend/src/pages/modelRegistry/screens/ModelLabels.tsx

```diff
@@ -9,26 +9,20 @@ import {
   Text,
 } from '@patternfly/react-core';
 import React from 'react';
-import { RegisteredModel } from '~/concepts/modelRegistry/types';
+import { ModelVersion, RegisteredModel } from '~/concepts/modelRegistry/types';
 import useDebounceCallback from '~/utilities/useDebounceCallback';
-import { getRegisteredModelLabels } from './utils';
+import { getLabels } from './utils';
 
 // Threshold count to decide whether to choose modal or popover
 const MODAL_THRESHOLD = 4;
 
-type RegisteredModelLabelsProps = {
-  rmName: string;
-  customProperties: RegisteredModel['customProperties'];
+type ModelLabelsProps = {
+  name: string;
+  customProperties: RegisteredModel['customProperties'] | ModelVersion['customProperties'];
 };
 
-const RegisteredModelLabels: React.FC<RegisteredModelLabelsProps> = ({
-  rmName,
-  customProperties,
-}) => {
-  const rmLabels = React.useMemo(
-    () => getRegisteredModelLabels(customProperties),
-    [customProperties],
-  );
+const ModelLabels: React.FC<ModelLabelsProps> = ({ name, customProperties }) => {
+  const allLabels = React.useMemo(() => getLabels(customProperties), [customProperties]);
```

Comment:
Without `useMemo`, the component re-renders due to the use of the helper function `getLabels`, so its use here is necessary.

Reply:
The issue doesn't lie with getLabels, is because we are using it later in the `useEffect`, why are we using a useEffect here?, shouldne be easier to have something like this?

```suggestion
  const [isLabelModalOpen, setIsLabelModalOpen] = React.useState(false);
  const [searchValue, setSearchValue] = React.useState('');

  const allLabels = getLabels(customProperties);
  const filteredLabels = allLabels.filter(
    (label) => label && label.toLowerCase().includes(searchValue.toLowerCase()),
  );
```